lib/fibonacci.rb

```diff
@@ -1,8 +1,25 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n)
+# Space Complexity - O(n)
 # Hint, you may want a recursive helper method
 def fibonacci(n)
```

Comment:
This works, but you're keeping the entire list of fibonacci numbers for the whole length of the recursion.  Instead you should only keep the last 2 fibonacci numbers.